src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -65,26 +67,35 @@ public GameParser(final String mapName) {
     this.mapName = mapName;
   }
 
+  /**
+   * Parses the XML Document provided by the given InputStream into a DOM.
+   * 
+   * @return The root Element of the DOM
+   */
+  public synchronized Element parseDom(final InputStream stream)
+      throws SAXException {
+    Preconditions.checkNotNull(stream, "InputStream must be non-null!");
+    return getDocument(stream).getDocumentElement();
+  }
+
   /**
    * Parses a file into a GameData object.
-   *
-   * @param delayParsing Should we only parse the game name, notes, and playerlist? Normally this should be "false",
-   *        except for the game chooser which should use the user set preference.
    */
-  public synchronized GameData parse(final InputStream stream, final AtomicReference<String> gameName,
-      final boolean delayParsing)
+  public synchronized GameData parse(final InputStream stream, final AtomicReference<String> gameName)
       throws GameParseException, SAXException, EngineVersionException, IllegalArgumentException {
```

Comment:
Nit: Can probably remove the unchecked `IllegalArgumentException` from the exception specification.  Instead we should document under what conditions it can be thrown.

Actually, it looks like we changed that behavior.  Whereas previously, a `null` stream would trigger an IAE, it now triggers an NPE.  The NPE is probably the correct choice, so no Javadocs should be needed to state that explicitly, as that behavior should be assumed.